library/core/src/clone.rs

```diff
@@ -307,60 +410,60 @@ pub unsafe trait CloneToUninit {
     ///
     /// This function may panic. (For example, it might panic if memory allocation for a clone
     /// of a value owned by `self` fails.)
-    /// If the call panics, then `*dst` should be treated as uninitialized memory; it must not be
+    /// If the call panics, then `*dest` should be treated as uninitialized memory; it must not be
     /// read or dropped, because even if it was previously valid, it may have been partially
     /// overwritten.
     ///
-    /// The caller may also need to take care to deallocate the allocation pointed to by `dst`,
+    /// The caller may also need to take care to deallocate the allocation pointed to by `dest`,
     /// if applicable, to avoid a memory leak, and may need to take other precautions to ensure
     /// soundness in the presence of unwinding.
```

Comment:
```suggestion
    /// if applicable, to avoid a memory leak.
```

I think the current phrasing implies that deallocating is necessary to maintain soundness, but that doesn't seem like an actual requirement.